decentraland/unity-explorer
# Fix: systems priority gate

## What does this PR change?
There were 2 problems:
1. Wrong `Type` of the group was checked by `UpdateTransformSystem` - the one with `Sync` prefix, which is doesn't exists in `SystemGroupsUpdateGate`
2. Both `UpdateTransformSystem` and `SystemGroupsUpdateGate` was removing (closing) the gate for this respective group, so only first who did that will have 1 additional frame for it.

Solutions:
1. Fixed the `Type` issue and add a Test to prevent it.
2. Closing gate for the group now happens inside `OnSystemGroupUpdateFinished`

## How to test the changes?

1. Launch the explorer
2. Check Plaza Carousel works as previous
3. Check Goerli rotating platforms has no jittering behavior (i.e. works as currently on prod)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
